addons/mail/static/tests/translation/translation_test.js

```diff
@@ -0,0 +1,50 @@
+/* @odoo-module */
+
+import { click, start, startServer, contains } from "@mail/../tests/helpers/test_utils";
+
+const spanishText = "Al mal tiempo, buena cara.";
+const englishText = "To bad weather, good face.";
+const rpc = { value: englishText, source: "Spanish", error: null };
+
+QUnit.module("Google Cloud Translation");
+
+QUnit.test("Toggle display of original/translated version of chatter message", async (assert) => {
+    let requestCount = 0;
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["mail.message"].create({
+        model: "res.partner",
+        body: spanishText,
+        author_id: pyEnv.odoobotId,
+        res_id: partnerId,
+    });
+    const { env, openFormView } = await start({
+        mockRPC(route) {
+            if (route === "/discuss/message/translate") {
+                requestCount++;
+                return rpc;
+            }
+        },
+    });
+
+    // The feature should be disabled if no API key was provided (hasMessageTranslationFeature=false).
+    await openFormView("res.partner", partnerId);
+    await contains(".o-mail-Message");
+    await contains("button[aria-label='Translate']", 0);
+    env.services["mail.store"].hasMessageTranslationFeature = true;
+    await openFormView("res.partner", partnerId);
+    await contains("button[aria-label='Translate']");
+    await contains("button[aria-label='Revert']", 0);
+
+    // Click acts as a toogle affecting its appearence and the actual message content displayed.
+    await click("button[aria-label='Translate']");
```

Comment:
Empty newline is not useful.
In general they distract more because it gives the wrong impression code before/after are 2 tests rather than a single test.

Reply:
I'll keep that in mind and I removed newlines in test files.